spec/unit/equalizer/universal_spec.rb

```diff
@@ -146,6 +146,22 @@ def initialize(firstname, lastname)
         expect(instance.hash)
           .to eql([firstname, lastname, klass].hash)
       end
+
+      context 'when immutable' do
+        let(:instance) { double(firstname: 'Foo', lastname: 'Bar') }
+        let(:equalizer) { Dry::Equalizer.new(:firstname, :lastname, immutable: true) }
+
+        before do
+          instance.extend(equalizer)
+        end
+
+        it 'is memoized' do
+          expect(instance).to receive(:firstname).once
```

Comment:
In keeping with the idea of testing only the public API, you could define your `klass` with `attr_accessor` and assert something like this:

```ruby
initial_hash = instance.hash
instance.firstname = 'Changed'
expect(instance.hash).to eql(initial_hash)
```